core/src/main/scala/org/typelevel/vault/Vault.scala

```diff
@@ -38,18 +38,38 @@ final class Vault private (private val m: Map[Unique.Token, Locker]) {
   /**
    * Lookup the value of a key in this vault
    */
-  def lookup[A](k: Key[A]): Option[A] = Vault.lookup(k, this)
+  def lookup[A](k: LookupKey[A]): Option[A] = Vault.lookup(k, this)
+
+  /**
+   * Lookup the value of a key in this vault
+   */
+  private[vault] def lookup[A](k: Key[A]): Option[A] = lookup(k: LookupKey[A])
+
+  /**
+   * Insert a value for a given key. Overwrites any previous value.
+   */
+  def insert[A](k: InsertKey[A], a: A): Vault = Vault.insert(k, a, this)
 
   /**
    * Insert a value for a given key. Overwrites any previous value.
    */
-  def insert[A](k: Key[A], a: A): Vault = Vault.insert(k, a, this)
+  private[vault] def insert[A](k: Key[A], a: A): Vault = insert(k: InsertKey[A], a)
 
   /**
    * Checks whether this Vault is empty
    */
   def isEmpty: Boolean = Vault.isEmpty(this)
 
+  /**
+   * Delete a key from the vault
+   */
+  def delete[A](k: InsertKey[A]): Vault = Vault.delete(k, this)
+
+  /**
+   * Delete a key from the vault
+   */
+  def delete[A](k: LookupKey[A]): Vault = Vault.delete(k, this)
```

Comment:
Uh, what happens if we call this with `Key`?  Does it pick one from the linearization of `Key`?  It doesn't really matter, I guess?

Reply:
If we call this with `Key` it uses the existing overload for `Key` and ignores both of these.

Reply:
Oh, I overlooked that there's still a `delete(Key[A])`.